pkg/controller/infrastructure/infraflow/reconcile.go

```diff
@@ -1382,3 +1452,30 @@ func (c *FlowContext) getZone(item *awsclient.Subnet) *aws.Zone {
 func getZoneName(item *awsclient.Subnet) string {
 	return item.AvailabilityZone
 }
+
+func cidrSubnet(baseCIDR string, newPrefixLength int, index int) (string, error) {
+	_, ipNet, err := net.ParseCIDR(baseCIDR)
+	if err != nil {
+		return "", err
+	}
+
+	baseIP := ipNet.IP
+	maskSize, addrSize := ipNet.Mask.Size()
+
+	if newPrefixLength <= maskSize || newPrefixLength > addrSize {
+		return "", fmt.Errorf("invalid new prefix length")
+	}
+
+	subnetSize := big.NewInt(1)
+	subnetSize = subnetSize.Lsh(subnetSize, uint(addrSize-newPrefixLength))
+	offset := big.NewInt(int64(index))
+	offset = offset.Mul(offset, subnetSize)
+
+	ipInt := big.NewInt(0)
+	ipInt.SetBytes(baseIP)
+	ipInt = ipInt.Add(ipInt, offset)
```

Comment:
Why not shorter?

```suggestion
	ipInt := ipInt.Add((&big.Int{}).SetBytes(baseIP), offset)
```

Reply:
The shorter version would look like this
```
ipInt := big.NewInt(0).SetBytes(baseIP).Add(big.NewInt(0).SetBytes(baseIP), offset)
```
I think the long version is easier to read and understand.

Reply:
If you want to go for the long version please use more than one variable name. The result may be `ipInt`, but in between it is not.